compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELinalgFunctionOutlining.cpp

```diff
@@ -183,21 +161,9 @@ void AMDAIELinalgFunctionOutliningPass::runOnOperation() {
     if (failed(maybeFunc)) return WalkResult::interrupt();
     func::FuncOp func = maybeFunc.value();
 
-    // Create a call into the outlined function. The operands of the compute op
-    // might need to be cast to a different type to match the outlined function.
-    {
-      SmallVector<Value> castOperands;
-      castOperands.reserve(computeOp->getOperands().size());
-      rewriter.setInsertionPoint(computeOp);
-      Location loc = computeOp.getLoc();
-      for (auto iter : llvm::enumerate(computeOp->getOperands())) {
-        Type type = func.getArgumentTypes()[iter.index()];
-        Value cast =
-            rewriter.createOrFold<memref::CastOp>(loc, type, iter.value());
-        castOperands.push_back(cast);
-      }
-      rewriter.create<func::CallOp>(loc, func, castOperands);
-    }
+    rewriter.setInsertionPoint(computeOp);
+    rewriter.create<func::CallOp>(computeOp.getLoc(), func,
+                                  computeOp->getOperands());
```

Comment:
This is a nice simplification. But I'm trying to understand why it doesn't need `cast` any more?

Reply:
The cast was there to cast from a type with an offset, to a type without an offset. This PR changes the logic so that if a memref has an offset, then it cannot be outlined. So we never need to cast.